fog/sample-paykit/src/cached_tx_data/mod.rs

```diff
@@ -370,6 +401,45 @@ impl CachedTxData {
             log::error!(self.logger, "Fog view protocol error: {}", err);
         }
 
+        log::trace!(
+            self.logger,
+            "Adding {} missed blocks ranges to the missed block ranges queue",
+            new_missed_block_ranges.len(),
+        );
+
+        for missed_block_range in &new_missed_block_ranges {
+            log::trace!(
+                self.logger,
+                "Missed Block start: {}, end: {}",
+                missed_block_range.start_block,
+                missed_block_range.end_block,
+            );
+        }
+
+        self.missed_block_ranges.extend(new_missed_block_ranges);
+        let fog_common_block_ranges: Vec<fog_common::BlockRange> =
+            CachedTxData::convert_block_ranges(&self.missed_block_ranges);
+        match fog_block_client.get_missed_block_ranges(fog_common_block_ranges) {
+            Ok(block_response) => {
+                let tx_out_records_from_missed_blocks: Vec<TxOutRecord> =
+                    CachedTxData::create_tx_out_records(&block_response);
+                txo_records.extend(tx_out_records_from_missed_blocks);
+                let updated_missed_block_ranges =
+                    CachedTxData::calculate_updated_missed_block_ranges(
+                        &self.missed_block_ranges,
+                        &block_response.blocks.into_vec(),
+                    );
+                self.missed_block_ranges = updated_missed_block_ranges;
```

Comment:
Should I call "clear" here instead? I found this [discussion](https://users.rust-lang.org/t/what-is-a-more-efficient-way-to-clear-a-vec/40190).

Reply:
I don't think so, I think what you did is good
if you wanted to avoid a copy you could try something like making `calculate_updated_missed_block_ranges` take a mutable reference but the way you did it is fine